src/Surfnet/ServiceProviderDashboard/Application/Command/Entity/ResetOidcSecretCommand.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\Command\Entity;
+
+use InvalidArgumentException;
+use Surfnet\ServiceProviderDashboard\Application\Command\Command;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Service;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Attribute;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Contact;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\OidcGrantType;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Validator\Constraints as SpDashboardAssert;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @SpDashboardAssert\HasAttributes()
+ */
+class ResetOidcSecretCommand implements Command
+{
+    /**
+     * @var string
+     * @Assert\Uuid
+     */
+    private $id;
+
+    /**
+     * @var Service
+     */
+    private $service;
+
+    /**
+     * @var string
+     *
+     * @Assert\NotBlank()
+     * @Assert\Choice(choices = {"production", "test"}, strict=true)
+     */
+    private $environment = Entity::ENVIRONMENT_TEST;
+
+    /**
+     * @var string
+     */
+    private $clientSecret;
+
+    /**
+     * @var string
+     */
+    private $manageId;
+
+    /**
+     * @param string $manageId
+     * @param string $environment
+     * @param Service $service
+     */
+    public function __construct($manageId, $environment, Service $service)
+    {
+        $this->id = null;
+        $this->service = $service;
+        $this->manageId = $manageId;
+
+        if (!in_array($environment, [
+            Entity::ENVIRONMENT_TEST,
+            Entity::ENVIRONMENT_PRODUCTION,
+        ])) {
+            throw new InvalidArgumentException(
+                "Unknown environment '{$environment}'"
+            );
+        }
+
+        $this->environment = $environment;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param string $id
+     */
+    public function setId($id)
```

Comment:
Try to make the command readonly and inject the ID in the constructor

Reply:
Fixed